chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -255,66 +345,73 @@ impl PeerManagerActor {
+            if can_prune_edges { Prune::PruneOncePerHour } else { Prune::Disable },
+            DELETE_PEERS_AFTER_TIME,
+        );
+
+        near_performance_metrics::actix::run_later(
+            ctx,
+            UPDATE_ROUTING_TABLE_INTERVAL,
+            move |act, ctx| {
+                act.update_routing_table_trigger(ctx);
+            },
+        );
+    }
+
     /// Receives list of edges that were verified, in a trigger every 20ms, and adds them to
     /// the routing table.
     fn broadcast_edges_trigger(&mut self, ctx: &mut Context<PeerManagerActor>) {
-        let me = self.my_peer_id.clone();
-
         let start = Instant::now();
         let mut new_edges = Vec::new();
         while let Some(edge) = self.routing_table_exchange_helper.edges_to_add_receiver.pop() {
-            if let Some(cur_edge) =
-                self.routing_table.get_edge(edge.peer0.clone(), edge.peer1.clone())
-            {
-                if cur_edge.nonce >= edge.nonce {
-                    // We have newer update. Drop this.
-                    continue;
-                }
-            }
-            // Add new edge update to the routing table.
-            self.add_verified_edges_to_routing_table(ctx, vec![edge.clone()]);
-            if let Some(other) = edge.other(&me) {
-                // We belong to this edge.
-                if self.active_peers.contains_key(&other) {
-                    // This is an active connection.
-                    match edge.edge_type() {
-                        EdgeType::Added => {}
-                        EdgeType::Removed => {
-                            // Try to update the nonce, and in case it fails removes the peer.
-                            self.try_update_nonce(ctx, edge.clone(), other);
-                            continue;
-                        }
-                    }
-                } else {
-                    match edge.edge_type() {
-                        EdgeType::Added => {
-                            self.wait_peer_or_remove(ctx, edge.clone());
-                            continue;
-                        }
-                        EdgeType::Removed => {}
-                    }
-                }
-            }
             new_edges.push(edge);
             if start.elapsed() >= BROAD_CAST_EDGES_MAX_WORK_ALLOWED {
                 break;
             }
         }
 
-        let new_data = SyncData { edges: new_edges, accounts: Default::default() };
-
-        if !new_data.is_empty() {
-            #[cfg(not(feature = "test_features"))]
-            let condition = true;
-            #[cfg(feature = "test_features")]
-            let condition = !self.adv_disable_edge_propagation;
-
-            if condition {
-                self.broadcast_message(
-                    ctx,
-                    SendMessage { message: PeerMessage::RoutingTableSync(new_data) },
-                )
+        if !new_edges.is_empty() {
+            for edge in new_edges.iter() {
```

Comment:
can you comment what this whole for loop does?

It seems to be some sort of reconciliation mechanism - we're looking at the new edges that are added to the system, focusing only on the ones that are pointing to us - and we try to match this to our internal state.

Reply:
Thanks, I added a comment:

```
            // Check whenever there is an edge indicating whenever there is a peer we should be
            // connected to but we aren't. And try to resolve the inconsistency.
            //
            // Also check whenever there is an edge indicating that we should be disconnected
            // from a peer, but we are connected. And try to resolve the inconsistency.
```